addons/binding/org.openhab.binding.harmonyhub/src/main/java/org/openhab/binding/harmonyhub/internal/handler/HarmonyHubHandler.java

```diff
@@ -83,54 +86,96 @@
-    private @Nullable HarmonyClient client;
+    private @NonNullByDefault({}) HarmonyClient client;
     private @Nullable ScheduledFuture<?> retryJob;
     private @Nullable ScheduledFuture<?> heartBeatJob;
 
-    private @Nullable HarmonyClientListener harmonyClientListener;
-
     private int heartBeatInterval;
 
     public HarmonyHubHandler(Bridge bridge, HarmonyHubHandlerFactory factory) {
         super(bridge);
         this.factory = factory;
+        client = new HarmonyClient();
+        client.addListener(this);
     }
 
     @Override
     public void handleCommand(ChannelUID channelUID, Command command) {
         logger.trace("Handling command '{}' for {}", command, channelUID);
 
-        HarmonyClient localClient = client;
-        if (localClient == null) {
-            logger.warn("Cannot send command '{}' on {} because HarmonyClient is null", command, channelUID);
+        if (!client.isConnected()) {
+            logger.warn("Cannot send command '{}' on {} because HarmonyClient is not connected", command, channelUID);
             return;
         }
 
         if (command instanceof RefreshType) {
-            localClient.getCurrentActivity().thenAccept(activity -> {
+            client.getCurrentActivity().thenAccept(activity -> {
                 updateState(activity);
             });
-        } else if (command instanceof StringType) {
-            try {
-                try {
-                    int actId = Integer.parseInt(command.toString());
-                    localClient.startActivity(actId);
-                } catch (NumberFormatException ignored) {
-                    localClient.startActivityByName(command.toString());
+            return;
+        }
+
+        Channel channel = getThing().getChannel(channelUID.getId());
+        if (channel == null) {
+            logger.warn("No such channel for UID {}", channelUID);
+            return;
+        }
+
+        switch (channel.getUID().getId()) {
+            case CHANNEL_CURRENT_ACTIVITY:
+                if (command instanceof DecimalType) {
+                    try {
+                        client.startActivity(((DecimalType) command).intValue());
+                    } catch (Exception e) {
+                        logger.error("Could not start activity", e);
+                    }
+                } else {
+                    try {
+                        try {
+                            int actId = Integer.parseInt(command.toString());
+                            client.startActivity(actId);
+                        } catch (NumberFormatException ignored) {
+                            client.startActivityByName(command.toString());
+                        }
+                    } catch (IllegalArgumentException e) {
+                        logger.warn("Activity '{}' is not known by the hub, ignoring it.", command);
+                    } catch (Exception e) {
+                        logger.error("Could not start activity", e);
```

Comment:
Using warn here seems more appropriate. It would also help to add the activity ID to the log message. According to the [logging guidelines](https://www.openhab.org/docs/developer/development/guidelines.html#e-logging):

> `error` logging should only be used to inform the user that something is tremendously wrong in his setup, the system cannot function normally anymore, and there is a need for immediate action. It should also be used if some code fails irrecoverably and the user should report it as a severe bug.

Reply:
👍